closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityProperties.java

```diff
@@ -0,0 +1,707 @@
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Load restricted security properties.
+     */
+    private void initProperties() {
+
+        if (debug != null) {
+            debug.println("Loading restricted security properties.");
+        }
+
+        descName = parseProperty(securityProps.getProperty(propsPrefix + ".desc.name"));
+        descNumber = parseProperty(securityProps.getProperty(propsPrefix + ".desc.number"));
+        descPolicy = parseProperty(securityProps.getProperty(propsPrefix + ".desc.policy"));
+        descSunsetDate = parseProperty(securityProps.getProperty(propsPrefix + ".desc.sunsetDate"));
+
+        jdkTlsDisabledNamedCurves = parseProperty(
+                securityProps.getProperty(propsPrefix + ".tls.disabledNamedCurves"));
+        jdkTlsDisabledAlgorithms = parseProperty(
+                securityProps.getProperty(propsPrefix + ".tls.disabledAlgorithms"));
+        jdkTlsDphemeralDHKeySize = parseProperty(
+                securityProps.getProperty(propsPrefix + ".tls.ephemeralDHKeySize"));
+        jdkTlsLegacyAlgorithms = parseProperty(
+                securityProps.getProperty(propsPrefix + ".tls.legacyAlgorithms"));
+        jdkCertpathDisabledAlgorithms = parseProperty(
+                securityProps.getProperty(propsPrefix + ".jce.certpath.disabledAlgorithms"));
+        jdkSecurityLegacyAlgorithm = parseProperty(
+                securityProps.getProperty(propsPrefix + ".jce.legacyAlgorithms"));
+        keyStoreType = parseProperty(
+                securityProps.getProperty(propsPrefix + ".keystore.type"));
+        keyStore = parseProperty(
+                securityProps.getProperty(propsPrefix + ".javax.net.ssl.keyStore"));
+
+        jdkSecureRandomProvider = parseProperty(
+                securityProps.getProperty(propsPrefix + ".securerandom.provider"));
+        jdkSecureRandomAlgorithm = parseProperty(
+                securityProps.getProperty(propsPrefix + ".securerandom.algorithm"));
+
+        if (debug != null) {
+            debug.println("Loaded restricted security properties.");
+        }
+    }
+
+    /**
+     * Load security constraints with type, algorithm, attributes.
+     *
+     * Example:
+     * RestrictedSecurity1.jce.provider.1 = SUN [{CertPathBuilder, PKIX, *}, {Policy,
+     * JavaPolicy, *}, {CertPathValidator, *, *}].
+     */
+    private void initConstraints() {
+
+        for (int pNum = 1; pNum <= providersSN.size(); pNum++) {
+
+            String providerName = providersSN.get(pNum - 1);
+            String providerInfo = securityProps
+                    .getProperty(propsPrefix + ".jce.provider." + pNum);
+
+            if (debug != null) {
+                debug.println("Loading constraints for security provider: " + providerName);
+            }
+
+            // Remove all the space.
+            providerInfo = providerInfo.trim().replaceAll(" ", "");
```

Comment:
I think we want more sophisticated parsing; there may be other kinds of whitespace, for example, and we should be providing better feedback when the string is not well-formed.

Reply:
@keithc-ca 
Add more checks in the methods initConstraints() to provide better feedback when the string is not well-formed.

Reply:
@keithc-ca 

I changed the removal of all whitespaces, to just remove the whitespaces in the format separator if there are, like "[{", "}]" and "},{". So I can then use this separator to get the constraint by splitting the string.